Sources/SparkConnect/SparkConnectClient.swift

```diff
@@ -0,0 +1,228 @@
+  /// - Parameter key: A string for key to look up.
+  /// - Returns: A string for the value of the key.
+  func getConf(_ key: String) async throws -> String {
+    try await withGRPCClient(
+      transport: .http2NIOPosix(
+        target: .dns(host: self.host, port: self.port),
+        transportSecurity: .plaintext
+      )
+    ) { client in
+      let service = SparkConnectService.Client(wrapping: client)
+      var request = getConfigRequestGet(keys: [key])
+      request.clientType = clientType
+      request.userContext = userContext
+      request.sessionID = self.sessionID!
+      let response = try await service.config(request)
+      return response.pairs[0].value
+    }
+  }
+
+  /// Create a ``ConfigRequest`` for `GetAll` operation.
+  /// - Returns: <#description#>
+  func getConfigRequestGetAll() -> ConfigRequest {
+    var request = ConfigRequest()
+    request.operation = ConfigRequest.Operation()
+    let getAll = ConfigRequest.GetAll()
+    request.operation.opType = .getAll(getAll)
+    return request
+  }
+
+  /// Request the server to get all configurations.
+  /// - Returns: A map of key-value pairs.
+  func getConfAll() async throws -> [String: String] {
+    try await withGRPCClient(
+      transport: .http2NIOPosix(
+        target: .dns(host: self.host, port: self.port),
+        transportSecurity: .plaintext
+      )
+    ) { client in
+      let service = SparkConnectService.Client(wrapping: client)
+      var request = getConfigRequestGetAll()
+      request.clientType = clientType
+      request.userContext = userContext
+      request.sessionID = self.sessionID!
+      let response = try await service.config(request)
+      var map = [String: String]()
+      for pair in response.pairs {
+        map[pair.key] = pair.value
+      }
+      return map
+    }
+  }
+
+  /// Create a `Plan` instance for `Range` relation.
+  /// - Parameters:
+  ///   - start: A start of the range.
+  ///   - end: A end (exclusive) of the range.
+  ///   - step: A step value for the range from `start` to `end`.
+  /// - Returns: <#description#>
+  func getPlanRange(_ start: Int64, _ end: Int64, _ step: Int64) -> Plan {
+    var range = Range()
+    range.start = start
+    range.end = end
+    range.step = step
+    var relation = Relation()
+    relation.range = range
+    var plan = Plan()
+    plan.opType = .root(relation)
+    return plan
+  }
+
+  /// Create a ``ExecutePlanRequest`` instance with the given session ID and plan.
+  /// The operation ID is created by UUID.
+  /// - Parameters:
+  ///   - sessionID: A string for the existing session ID.
+  ///   - plan: A plan to execute.
+  /// - Returns: An ``ExecutePlanRequest`` instance.
+  func getExecutePlanRequest(_ sessionID: String, _ plan: Plan) async
+    -> ExecutePlanRequest
+  {
+    var request = ExecutePlanRequest()
+    request.clientType = clientType
+    request.userContext = userContext
+    request.sessionID = sessionID
```

Comment:
Shouldn't be `self.sessionID`?

Reply:
You are correct, @viirya . Let me fix this.